CONTRIBUTING.md

```diff
@@ -0,0 +1,58 @@
+# Contributing to Cranelift
+
+## Welcome!
+
+Cranelift is a very ambitious project with many goals, and while we're
+confident we can achieve some of them, we see many opportunities for people
+to get involved and help us achieve even more.
+
+### Ask questions! Give feedback!
+
+This is a relatively young project, and not everything we hope to do with it
+is reflected in the code or documentation yet. If you see things that seem
+missing or that don't make sense, or even that just don't work the way you
+expect them to, we're interested to hear about it!
+
+### Mentoring
+
+We're happy to mentor people, whether you're learning Rust, learning about
+compiler backends, learning about machine code, learning about how Cranelift
+does things, or all together at once.
+
+We tag issues in the issue tracker marked "good first issue" when we can, so
+that's sometimes a good place to get started. Also, we encourage people to just
+look around and find things they're interested in. This a good time to get
+involved, as there aren't a lot of things set in stone yet.
+
+## Coding Guidelines
+
+For the most part, Cranelift follows common Rust conventions, though we do
+have a few additional things to be aware of.
+
+### rustfmt
+
+All PRs must be formatted according to the current stable rustfmt, and this is
+checked in the continuous integration tests. See
+[here](https://github.com/rust-lang-nursery/rustfmt#quick-start) for setup.
```

Comment:
Is it a good place to mention the `format-all.sh` script?

Reply:
Mentioned.